lib/elixir/src/elixir_env.erl

```diff
@@ -138,10 +139,14 @@ is_compiler_var([$_]) -> true;
 is_compiler_var([Var | Rest]) when Var =:= $_; Var >= $A, Var =< $Z -> is_compiler_var(Rest);
 is_compiler_var(_) -> false.
 
-format_error({unused_var, Name}) ->
-  case atom_to_list(Name) of
-    "_" ++ _ ->
-      io_lib:format("unknown compiler variable \"~ts\" (expected one of __MODULE__, __ENV__, __DIR__, __CALLER__, __STACKTRACE__)", [Name]);
-    _ ->
-      io_lib:format("variable \"~ts\" is unused (if the variable is not meant to be used, prefix it with an underscore)", [Name])
-  end.
+format_error({unused_var, Name, VariableWithSameNameExistsInContext}) when is_atom(Name) ->
+  format_error({unused_var, atom_to_list(Name), VariableWithSameNameExistsInContext});
```

Comment:
Instead of calling yourself recursively, I think a `case` on the result of `atom_to_list` would be clearer. You can match on `_VariableWithSameNameExistsInContext` in a guard, like this:

    Name when VariableWithSameNameExistsInContext ->

I also think `VariableWithSameNameExistsInContext` is slightly wrong. We should probably call it `Overridden`?

Reply:
Done